proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,636 @@
+    uninitialized_copy(Rng&& rng, O result);
+template <InputIterator I, ForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_copy_n(I first, difference_type_t<I> n, O result);
+template <InputIterator I, Sentinel<I> S, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)> uninitialized_move(I first, S last, O result);
+template <InputRange Rng, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<iterator_t<Rng>>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
+    uninitialized_move(Rng&& rng, O result);
+template <InputIterator I, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_move_n(I first, difference_type_t<I> n, O result);
+template <ForwardIterator I, Sentinel<I> S, class T>
+  requires Constructible<value_type_t<I>, const T&>() &&
+           __ReferenceTo<I, T>()
+  I uninitialized_fill(I first, S last, const T& x);
+template <ForwardRange Rng, class T>
+  requrires Constructible<value_type_t<iterator_t<Rng>>, const T&>() &&
+            __ReferenceTo<iterator_t<Rng>, T>()
+  safe_iterator_t<I>
+  uninitialized_fill(Rng&& rng, const T& x);
+template <ForwardIterator I, class T>
+  requires Constructible<value_type_t<I>, const T&>() &&
+           __ReferenceTo<I, T>()
+  I uninitialized_fill_n(I first, difference_type_t<I> n, const T& x);
+template <Destructible T>
+  void destroy_at(T* location) noexcept;
+template <InputIterator I, Sentinel<I> S>
+  requires Destructible<value_type_t<I>>() &&
+           __ReferenceTo<iterator_t<Rng, value_type_t<Rng>>>()
+    I destroy(I first, S last) noexcept;
+template <InputRange Rng>
+  requires Destructible<value_type_t<Rng>>() &&
+           __ReferenceTo<iterator_t<Rng>, value_type_t<iterator_t<Rng>>>()
+    safe_iterator_t<Rng> destroy(Rng&& rng) noexcept;
+template <InputIterator I>
+  requires Destructible<value_type_t<I>>() &&
+           __ReferenceTo<I, value_type_t<I>>()
+    I destroy_n(I first, difference_type_t<I> n) noexcept;
+
+template <class I, class T>
+concept bool __ReferenceTo() { // exposition only
+  return InputIterator<I>() &&
+    is_lvalue_reference_v<reference_t<I>> &&
+    Same<remove_cv_t<remove_reference_t<reference_t<I>>>, T>();
+}
+\end{codeblock}
+} %% \color{addclr}
+
+\setcounter{subsection}{9}
+\rSec2[specialized.algorithms]{Specialized algorithms}
+This section assumes the definition of a \tcode{NoThrowForwardIterator} concept, which shall be described in a sibling
+proposal as an addition to \tcode{<experimental/ranges/iterator>}.
```

Comment:
>```cpp
template <class I>
concept bool @\xname{NoThrowForwardIterator}@() { // @\expos@
  return ForwardIterator();
}
```
 ... give a short description in a paragraph here (wording largely stolen from N4606)

Is it safe to provide an unchecked-at-compile-time concept? I thought one of the advantages of using an stl2 entity over its `std`-equivalent is because all the requirements are enforced at compile-time.